### PR TITLE
KAFKA-5995; Rename AlterReplicaDir to AlterReplicaDirs

### PR DESCRIPTION
**Pull request by lindong28**

### CODE REVIEW

**Comment by lindong28** [comment]
This patch made the following changes:

- Renamed AdminClient.alterReplicaDir to AdminClient.alterReplicaDirs. And rename the corresponding request, response, option, result, API key and test classes similarly.
- Renamed AdminClient.describeReplicaLogDir to AdminClient.describeReplicaLogDirs. And rename the corresponding option, result and test classes similarly.

The difference between log dir and dir is that, dir of a partition is generally subdirectory in the log dir. Regarding the choice between dir and logDir, my understanding of the guideline is that we should use logDir if it is wrong to interrupt it as dir. For example, `DirNotFoundException` would be misleading if it is actually the log directory that is not found.

One remaining question is whether we should rename alterReplicaDirs to alterReplicaLogDirs. In this case I am inclined to use alterReplicaDirs because it is short and we are actually changing both dir and log dir of the replica. The first field in AlterReplicaDirsRequest is `log_dirs`. Thus it is clear to developer that which value should be provided in this field. For users of AdminClient, we have specified in the Java doc that replicaAssignment is `The replicas with their log directory absolute path`.

Ping @junrao @ijuma for comment.

**Comment by lindong28** [comment]
I validated that all necessary changes have been made by using intellij to search for symbol with names alterReplicaDir and describeReplicaLogDir. This intellij feature allows us to find all classes, methods and public variables with this name prefix. It won't catch function local variable but function local variable can be renamed in the future.

**Comment by junrao** [comment]
@lindong28 : Thanks for the patch. It seems that we should be consistent btw alterReplicaDirs() and describeReplicaLogDirs() too. Using logDirs seems better since it's consistent with describeLogDirs(). So how about alterReplicaLogDirs() and describeReplicaLogDirs()?

**Comment by lindong28** [comment]
@junrao Sure. I have updated the patch as suggested.

File: clients/src/main/java/org/apache/kafka/common/protocol/ApiKeys.java [hunk 1]
```diff
@@ -164,8 +164,8 @@ public Struct parseResponse(short version, ByteBuffer buffer) {
             DescribeConfigsResponse.schemaVersions()),
     ALTER_CONFIGS(33, "AlterConfigs", AlterConfigsRequest.schemaVersions(),
             AlterConfigsResponse.schemaVersions()),
-    ALTER_REPLICA_DIR(34, "AlterReplicaDir", AlterReplicaDirRequest.schemaVersions(),
-            AlterReplicaDirResponse.schemaVersions()),
+    ALTER_REPLICA_LOG_DIRS(34, "AlterReplicaDirs", AlterReplicaLogDirsRequest.schemaVersions(),
```
Comment (junrao) [inline clients/src/main/java/org/apache/kafka/common/protocol/ApiKeys.java]: AlterReplicaDirs => AlterReplicaLogDirs ?

**Review by junrao — COMMENTED** [review]
@lindong28 : Thanks for the patch. LGTM. Just one more comment below.

File: clients/src/main/java/org/apache/kafka/common/protocol/ApiKeys.java (hunk 1 above)
Comment (lindong28) [reply clients/src/main/java/org/apache/kafka/common/protocol/ApiKeys.java]: Sorry, I should have noticed this. Thanks so much for your detailed review!

Comment (lindong28) [reply clients/src/main/java/org/apache/kafka/common/protocol/ApiKeys.java]: I have updated the patch to fix it.

**Comment by junrao** [comment]
@lindong28 : Thanks for the patch. LGTM.

**Pull request closed without merging**